### PR TITLE
Add sigs.k8s.io/structured-merge-diff jobs

### PR DESCRIPTION
**Pull request by mvladev**
fixes kubernetes-sigs/structured-merge-diff#7

### CODE REVIEW

File: config/jobs/kubernetes-sigs/structured-merge-diff/structured-merge-diff-ci.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+periodics:
+- name: ci-structured-merge-diff-test
+  interval: 1h
+  labels:
+    preset-service-account: "true"
+  spec:
```
Comment (BenTheElder) [inline config/jobs/kubernetes-sigs/structured-merge-diff/structured-merge-diff-ci.yaml]: For this job please add `decorate: true` above the `spec` field, and this to specify the checkout:
```
extra_refs:
- org: kubernetes-sigs
  repo: structured-merge-diff
  base_ref: master
  path_alias: sigs.k8s.io/structured-merge-diff
```
Then make the args something like:
```
command:
- go
args:
- test
- ./...
```

File: config/jobs/kubernetes-sigs/structured-merge-diff/structured-merge-diff-presubmits.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+presubmits:
+  kubernetes-sigs/structured-merge-diff:
+  - name: pull-structured-merge-diff-test
+    always_run: true
+    labels:
+      preset-service-account: "true"
+    spec:
```
Comment (BenTheElder) [inline config/jobs/kubernetes-sigs/structured-merge-diff/structured-merge-diff-presubmits.yaml]: For this job please add `decorate: true` above the `spec` field followed by: `path_alias: sigs.k8s.io/structured-merge-diff`
Then make the args something like:
```
command:
- go
args:
- test
- ./...
```

**Comment by BenTheElder** [comment]
we have some much more integrated tooling for new jobs (https://github.com/kubernetes/test-infra/blob/master/prow/pod-utilities.md), left comments about how to fix the job definitions for these.

File: config/jobs/kubernetes-sigs/structured-merge-diff/structured-merge-diff-presubmits.yaml (hunk 2 above)
Comment (mvladev) [reply config/jobs/kubernetes-sigs/structured-merge-diff/structured-merge-diff-presubmits.yaml]: Do I need to add `extra_refs` here as well?

**Comment by mvladev** [comment]
Thanks for the suggestions @BenTheElder ! Those issues are (hopefully) resolved in 47a989a.

File: config/jobs/kubernetes-sigs/structured-merge-diff/structured-merge-diff-presubmits.yaml (hunk 2 above)
Comment (BenTheElder) [reply config/jobs/kubernetes-sigs/structured-merge-diff/structured-merge-diff-presubmits.yaml]: nope, presubmits know what repo and ref to checkout based on the commit they're testing in the PR, other jobs need to be told what you want, you _could_ add extra_refs here if you needed to check out another repo as well for some reason (some jobs do)

**Comment by BenTheElder** [comment]
Ah the remaining part is the most clunky unfortunately:
The CI job (`ci-structured-merge-diff-test`) needs to be in a testgrid dashboard in testgrid/config.yaml

There are some docs in testgrid/README.md, but you can also look at eg how https://k8s-testgrid.appspot.com/sig-api-machinery-kube-storage-version-migrator is configure din that file

**Comment by mvladev** [comment]
@BenTheElder thanks again. All tests pass now.

File: config/jobs/kubernetes-sigs/structured-merge-diff/structured-merge-diff-ci.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+periodics:
+- name: ci-structured-merge-diff-test
+  interval: 1h
```
Comment (BenTheElder) [inline config/jobs/kubernetes-sigs/structured-merge-diff/structured-merge-diff-ci.yaml]: QQ: do we expect this to change often enough & have flaky enough tests to warrant testing every hour? Seems like we should reduce this or switch to a postsubmit

**Comment by BenTheElder** [comment]
thanks, sorry for the delayed response, one question about the CI interval

File: config/jobs/kubernetes-sigs/structured-merge-diff/structured-merge-diff-ci.yaml (hunk 3 above)
Comment (BenTheElder) [reply config/jobs/kubernetes-sigs/structured-merge-diff/structured-merge-diff-ci.yaml]: placed a `/hold` if you think this is reasonable, go for `/hold cancel`. I suspect it's a bit high, but it shouldn't be a problem :-)

**Comment by BenTheElder** [comment]
/lgtm
Please file PRs to adjust as needed! Thanks for working on this :-)
You should see results on testgrid fairly shortly after merging this.

**Pull request merged**